core/src/wallet/ripple/api_impl/RippleLikeTransactionApi.cpp

```diff
@@ -43,6 +43,38 @@
 
 namespace ledger {
     namespace core {
+        // a helper to write VLE fields
+        bool writeLengthPrefix(BytesWriter& writer, uint64_t size) {
+            // encoding here: https://developers.ripple.com/serialization.html#length-prefixing
+            if (size <= 192) {
+                writer.writeByte(size & 0xFF);
+
+                return true;
+            } else if (size <= 12480) {
+                //     l       = 193 + ((byte1 - 193) * 256) + byte2
+                // <=> l - 193 = ((byte1 - 193) * 256) + byte2
+                // <=> byte2 = (l - 193) & 0xFF
+                // <=> byte1 = 193 + ((l - 193) >> 8) & 0xFF
+                size -= 193;
+                writer.writeByte(193 + (size >> 8) & 0xFF);
+                writer.writeByte(size & 0xFF);
+
+                return true;
+            } else if (size <= 918744) {
+                //     l         = 12481 + ((byte1 - 241) * 65536) + (byte2 * 256) + byte3
+                // <=> l - 12481 = ((byte1 - 241) * 65536) + (byte2 * 256) + byte3
+                // <=> byte3 = (l - 12481) & 0xFF
+                // <=> byte2 = ((l - 12481) >> 8) & 0xFF
+                // <=> byte1 = 241 + ((l - 12481) >> 16) & 0xFF
+                size -= 12481;
+                writer.writeByte(241 + (size >> 16) & 0xFF);
```

Comment:
According to https://en.cppreference.com/w/cpp/language/operator_precedence `&` has lower priority then +, so the result will be (241 + (size>>16) ) & 0xFF

Reply:
Fixed.